### ISSUE TITLE
[Bug] Unwanted branches in repository

### ISSUE THREAD

**Issue by neteler** [OSGeo/grass#1286]
**Describe the bug**

At time we have 4 unwanted branches in master which I do not manage to remove (nor the author):

- hellik-patch-1
- hellik-patch-2
- hellik-patch-3
- gh_action_codeql

```
git push -d upstream hellik-patch-1
remote: error: GH006: Protected branch update failed for refs/heads/hellik-patch-1.
remote: error: Cannot delete this protected branch
To github.com:OSGeo/grass.git
 ! [remote rejected]     hellik-patch-1 (protected branch hook declined)
error: failed to push some refs to 'github.com:OSGeo/grass.git'

git push -d upstream gh_action_codeql
remote: error: GH006: Protected branch update failed for refs/heads/gh_action_codeql.
remote: error: Cannot delete this protected branch
To github.com:OSGeo/grass.git
 ! [remote rejected]     gh_action_codeql (protected branch hook declined)
error: failed to push some refs to 'github.com:OSGeo/grass.git'

...
```

Any idea how to proceed? Lift the branch protection for a moment (https://github.com/OSGeo/grass/settings/branches)?

If anyone with more Git(hub) experience could take over I'd be grateful (maybe @landam @wenzeslaus )

**Comment by wenzeslaus** [comment]
Previously, I just set there all branches to be protected because, given our workflows, there is no reason we should have any other branches in the main repo. Now I duplicated the rule, so there is one for release branches and one for master branch. (There is no way on GitHub to have same protection rule for branches with completely different names, so two rules are needed.)

The easiest way for managing branches with GitHub is probably the web interface here (although the message in the command line is equally clear):

https://github.com/OSGeo/grass/branches

I don't know about settings to prevent people from pushing random branches into the main repo instead of their fork, but even if I did, I would be reluctant to apply it as it may stand in the way in some valid use cases. 

A side note on concepts and terminology:

There are no branches in master. Saying repo[sitory] is probably sufficient in this context.

Additionally, I suggest to always say _master branch_ instead of just master to make it clear that it is indeed a branch.
